### PR TITLE
Use NSCAssert instead of assert

### PR DESCRIPTION
**Pull request by dfed**
cc @efirestone

### CODE REVIEW

File: thumborurl/ThumborURL.m [hunk 1]
```diff
@@ -285,7 +285,7 @@ @implementation NSURL (ThumborURL)
 
 + (id)TU_secureURLWithOptions:(TUOptions *)options imageURL:(NSURL *)imageURL baseURL:(NSURL *)baseURL securityKey:(NSString *)securityKey;
 {
-    assert(securityKey.length > 0);
+    NSAssert(securityKey.length > 0, @"securityKey required for calling %@", NSStringFromSelector( _cmd));
```
Comment (pwesten) [inline thumborurl/ThumborURL.m]: No need for `_cmd`, NSAssert does this for you.

Comment (dfed) [inline thumborurl/ThumborURL.m]: Yeah. Was keeping the style of an existing NSAssert. Will remove it here (and there)

**Comment by pwesten** [comment]
LGTM.

**Comment by mthole** [comment]
@dfed What's the "why" for this change? Warnings fix?

**Comment by dfed** [comment]
@mthole yup. In RELEASE these asserts were compiled out, making a few variables unused, which caused a build error. We're insanely confident that these asserts will never be hit, so we're willing to put them into RELEASE builds.

**Pull request merged**